tests/integration_tests/core/test_lock_store.py

```diff
@@ -0,0 +1,194 @@
+
+        async with self.lock_store.lock(
+            message.sender_id, wait_time_in_seconds=lock_wait
+        ):
+            # hold up the message processing after the lock has been acquired
+            await asyncio.sleep(wait)
+
+            # write message to file as it's processed
+            with open(str(results_file), "a+") as f_1:
+                f_1.write(message.text + "\n")
+
+            return None
+
+    # We'll send n_messages from the same sender_id with different blocking times
+    # after the lock has been acquired.
+    # We have to ensure that the messages are processed in the right order.
+    with patch.object(Agent, "handle_message", mocked_handle_message):
+        # use decreasing wait times so that every message after the first one
+        # does not acquire its lock immediately
+        wait_times = np.linspace(0.1, 0.05, n_messages)
+        tasks = [
+            default_agent.handle_message(
+                UserMessage(f"sender {i}", sender_id="some id"), wait=k
+            )
+            for i, k in enumerate(wait_times)
+        ]
+
+        # execute futures
+        await asyncio.gather(*(asyncio.ensure_future(t) for t in tasks))
+
+        expected_order = [f"sender {i}" for i in range(len(wait_times))]
+
+        # ensure order of incoming messages is as expected
+        with open(str(incoming_order_file)) as f:
+            incoming_order = [line for line in f.read().split("\n") if line]
+            assert incoming_order == expected_order
+
+        # ensure results are processed in expected order
+        with open(str(results_file)) as f:
+            results_order = [line for line in f.read().split("\n") if line]
+            assert results_order == expected_order
+
+        # Every message after the first one will wait `lock_wait` seconds to acquire its
+        # lock (`wait_time_in_seconds` kwarg in `lock_store.lock()`).
+        # Let's make sure that this is not blocking and test that total test
+        # execution time is less than  the sum of all wait times plus
+        # (n_messages - 1) * lock_wait
+        time_limit = np.sum(wait_times[1:])
+        time_limit += (n_messages - 1) * lock_wait
+        assert time.time() - start_time < time_limit
+
+
+async def test_lock_error(default_agent: Agent):
+    lock_lifetime = 0.01
+    wait_time_in_seconds = 0.01
+    holdup = 0.5
+
+    # Mock message handler again to add a wait time holding up the lock
+    # after it's been acquired
+    async def mocked_handle_message(self, message: UserMessage) -> None:
+        async with self.lock_store.lock(
+            message.sender_id,
+            wait_time_in_seconds=wait_time_in_seconds,
+            lock_lifetime=lock_lifetime,
+        ):
+            # hold up the message processing after the lock has been acquired
+            await asyncio.sleep(holdup)
+
+        return None
+
+    with patch.object(Agent, "handle_message", mocked_handle_message):
```

Comment:
optional: let's use monkeypatch as in the other tests for consistency

Reply:
fixed in d910d53f948; I'm somewhat of a `unittest.patch` guy